sdk/core/Azure.Core/src/NullableResponseOfT.cs

```diff
@@ -40,6 +81,79 @@ public abstract class NullableResponse<T>
         public override int GetHashCode() => base.GetHashCode();
 
         /// <inheritdoc />
-        public override string ToString() => $"Status: {GetRawResponse()?.Status}, Value: {(HasValue ? Value : NoValue)}";
+        public override string ToString()
+             => $"Status: {GetRawResponse()?.Status}, Value: {(HasValue ? Value : NoValue)}";
```

Comment:
It appears that [this could throw](https://github.com/Azure/azure-sdk-for-net/pull/42255/files#diff-f6e6911f3651305a85cadb3221ab7a6cdd79f36aa4988a01690918908890e463R109) if response is `DefaultRawResponse`.  I think we should handle that so this never throws, if so.

Reply:
I would counter that if someone had created an instance of `NullableResponse<T>` with the DefaultRawResponse they had done it incorrectly and we _should_ throw so that someone could find and fix the error.  When would we want a `Response<T>` with an invalid return result from GetRawResponse?

Reply:
That makes sense if this is a contrived scenario - I was just thinking that generally we don't want to throw from `ToString()`

Reply:
Yes - any implementation that creates an instance of `DefaultResponse` is incorrect, so this should never throw in practice.